web/src/views/FileBrowserView/PublishFileBrowser.vue

```diff
@@ -166,11 +227,14 @@ export default {
           results: [asset],
         } = await publishRest.assets(identifier, version, { params: { path: relativePath } });
 
-        this.$set(
-          this.itemDownloads,
-          item.name,
-          publishRest.assetDownloadURI(identifier, version, asset),
-        );
+        this.$set(this.itemData, item.name, {
+          download: publishRest.assetDownloadURI(identifier, version, asset),
+          delete: {
+            identifier,
+            version,
+            uuid: asset.uuid,
+          },
```

Comment:
I don't see any particular reason to store a `delete` field here. For deletion, all you need is the identifier, the version, and the asset uuid. The identifier and version won't change anyway, so all you really need to store is the uuid. I'd propose something like this:

```suggestion
          uuid,
```

Then below, you would just get the `itemData` entry and look at the `uuid` field. I also think you could get rid of the `itemDownload` and `itemDelete` functions, and just access `this.itemData` and the respective fields when/where you need them.

Lmk what you think.

Reply:
Thanks for catching this. Still need `itemDownload` for convenience (because I still want to check for `undefined` since I don't really understand the type safety of these values), but check out 5074f5ef7da82b803beb9b5c13766b9b42044142 for your main suggestion.

Reply:
FWIW, this is the only place `itemDownload` is used:

```
<v-list-item-action v-if="itemDownload(item.name)">
  <v-btn
    icon
    :href="itemDownload(item.name)"
  >
  ...
```

My main point was that you could instead just do the following:

```
<v-list-item-action v-if="itemData[name]">
  <v-btn
    icon
    :href="itemData[name].download"
  >
  ...
```

The `download` field is always present on each `itemData` entry, so this wouldn't be an issue. However, it really doesn't matter to me, and what you've done addresses the core of my complaint, so I'll approve this. If you do happen to make the additional change I mentioned above (as well as removing the `itemDownload` function) anyway, my approval still stands.

Reply:
Thanks for clarifying. Check out 7cd32721dd912c0e10ba95e73560927434b727c5.